clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -1212,12 +1211,64 @@ scheduler_tmpl_t<RecordType, ReaderType>::read_traced_schedule()
     return STATUS_SUCCESS;
 }
 
+template <typename RecordType, typename ReaderType>
+typename scheduler_tmpl_t<RecordType, ReaderType>::scheduler_status_t
+scheduler_tmpl_t<RecordType, ReaderType>::remove_zero_instruction_segments(
+    std::vector<std::vector<schedule_input_tracker_t>> &input_sched,
+    std::vector<std::vector<schedule_output_tracker_t>> &all_sched)
+
+{
+    // For a cpuid pair with no instructions in between, our
+    // instruction-ordinal-based control points cannot model both sides.
```

Comment:
Can you remind me why our control points are not record-ordinal-based?

Reply:
The skip and region-of-interest features were designed thinking more about instructions, the more natural unit for microarchitectural simulators.  It seemed like that was much more usable for a user, and translated to other venues like PMU counts.  The scheduler replay features were also designed that way.  But, that makes the infrastructure messy as the underlying records are not built that way.  Xref #6716 on an instruction-based iterator.

I added this as a comment in the code.